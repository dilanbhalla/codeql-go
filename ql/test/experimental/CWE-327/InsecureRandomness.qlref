experimental/CWE-327/InsecureRandomness.ql
