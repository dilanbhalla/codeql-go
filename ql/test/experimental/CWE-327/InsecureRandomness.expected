edges
| InsecureRandomness.go:24:12:24:40 | call to New : pointer type | InsecureRandomness.go:27:25:27:29 | nonce |
| InsecureRandomness.go:24:12:24:40 | call to New : pointer type | InsecureRandomness.go:27:32:27:36 | nonce |
nodes
| InsecureRandomness.go:24:12:24:40 | call to New : pointer type | semmle.label | call to New : pointer type |
| InsecureRandomness.go:27:25:27:29 | nonce | semmle.label | nonce |
| InsecureRandomness.go:27:32:27:36 | nonce | semmle.label | nonce |
#select
| InsecureRandomness.go:27:25:27:29 | nonce | InsecureRandomness.go:24:12:24:40 | call to New : pointer type | InsecureRandomness.go:27:25:27:29 | nonce | $@ is a weak key used in cryptographic algorithm. | InsecureRandomness.go:24:12:24:40 | call to New | Random number |
| InsecureRandomness.go:27:32:27:36 | nonce | InsecureRandomness.go:24:12:24:40 | call to New : pointer type | InsecureRandomness.go:27:32:27:36 | nonce | $@ is a weak key used in cryptographic algorithm. | InsecureRandomness.go:24:12:24:40 | call to New | Random number |
